src/Sparrow.Server/Utils/DiskStatsGetter.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Sparrow.Logging;
+using Sparrow.Server.Platform.Posix;
+
+namespace Sparrow.Server.Utils
+{
+    public interface IDiskStatsGetter
+    {
+        DiskStatsResult Get(string drive);
+        Task<DiskStatsResult> GetAsync(string drive);
+    }
+
+    internal class DiskStatsGetter : IDiskStatsGetter
+    {
+        private readonly TimeSpan _minInterval;
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger("Server", typeof(DiskStatsGetter).FullName);
+
+        private readonly ConcurrentDictionary<string, Task<PrevInfo>> _previousInfo = new ConcurrentDictionary<string, Task<PrevInfo>>();
+
+        public DiskStatsGetter(TimeSpan minInterval)
+        {
+            _minInterval = minInterval;
+        }
+
+        public DiskStatsResult Get(string drive) => GetAsync(drive).ConfigureAwait(false).GetAwaiter().GetResult();
+        
+        public async Task<DiskStatsResult> GetAsync(string drive)
+        {
```

Comment:
shouldn't we assume that this can throw? what then? what are we doing in cpu calc, etc?